web/modules/custom/rdf_ff/src/SchemaFieldValidatorInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Drupal\rdf_ff;
+
+/**
+ * Interface SchemaFieldValidatorInterface.
+ */
+interface SchemaFieldValidatorInterface {
+
+  /**
+   * Checks if a field is defined in the ontology of it's class.
+   *
+   * @param string $entity_type_id
+   *   The entity type id of the class defined in the ontology.
+   * @param string $bundle
+   *   The bundle of the class defined in the ontology.
+   * @param string $field_name
+   *   The field name.
+   * @param string $column
+   *   The column name.
+   *
+   * @return bool
+   *   Whether the field name is defined in the schema.
+   */
+  public function isDefinedInSchema($entity_type_id, $bundle, $field_name, $column = '');
```

Comment:
Well we the `\Drupal\rdf_entity\RdfFieldHandler::getFieldPredicates` method already does that. If no column is passed, it fetches the main property.
We don't need actually a per field check because the field is not part of the schema. Only the columns are.

Maybe we need something like that when the blank nodes are implemented but I think it will take some time for that to happen :/

Reply:
Also, isDefinedInSchema is better now that I have renamed the module :D

Reply:
> We don't need actually a per field check because the field is not part of the schema.

This issue is about providing an API to "Identify if an individual field is an ADMS field" (issue title). Moreover, as I understood, we want, in a future ticket, to block the edit of such fields. But from a Drupal perspective the unit is the field. The access is granted on a field level, not a column level. Meaning that, when we'll have to implement the per-field access, we'll have to do that logic there (iterating over columns or picking the main column). But as I told, this is the scope of this ticket: "Give me entity type, bundle and field name and I'll tell you if it's an ADMS field". Still think that we need a per-field check in order to support the future ticket.

Reply:
The problem with what you are saying is the definition of the terms.
We are bridging the two database systems but the idea does not change.

In drupal we have fields. These fields have properties. For example, the text field has the properties `value`, `format`, the integer field has the `value`, `prefix`, `suffix` etc.
These are stored in the `field_data_field_name` table. 

However, in the triplestore, we do not have such thing. We have blank nodes but they are not supported at the moment, and even if they were, we do not map properties to fields(this is the reason why we do not support deltas at the moment).
So a graph triple will not contain a field value, which is a group of properties, rather than a single column of the field.

In addition to that, we have made some compromises to manage and store the complexity of drupal fields to the triplestore. An example of that is that for text fields, the language is stored as a suffix. E.g., the English version of a French word "Liberation" will be stored as `"Liberty"@en` because in triplestore, the language for strings, is stored as a suffix.
For other cases, it might be an independent property.

For these reasons, we cannot track a field as an entity but we have to track each property separately.
The solution to that issue comes by the fact that we take care and store each main adms field into a different Drupal field. So we do not have 2 ADMS fields as properties of the same Drupal field.

Later on, when we need to check if a field of an entity type and a bundle are part of the ADMS schema, we need to call the method `::isDefinedInSchema` without a column passed. The method will then use the main property as a value which can include a predicate included in the ADMS.

In conclusion, we do not store a relation between a field and a schema value. If you see any of the field.storage related to the rdf entities or even the rdf_entity mappings that you created earlier, none of the predicates are mapped to a field or an entity in general. They are mapped to properties.

The functionality is meant to check if a Drupal property (whether a base field or an extra field property) matches an ADMS defined property. 
If a property is not passed, the default property is used. Currently the default properties are these that we use to match the ADMS fields.

Reply:
OK. Let's go